pyiceberg/io/pyarrow.py

```diff
@@ -1715,53 +1715,65 @@ def fill_parquet_file_metadata(
 
 
 def write_file(table: Table, tasks: Iterator[WriteTask]) -> Iterator[DataFile]:
-    task = next(tasks)
+    # task = next(tasks)
+
+    # try:
+    #     _ = next(tasks)
+    #     # If there are more tasks, raise an exception
+    #     raise NotImplementedError("Only unpartitioned writes are supported: https://github.com/apache/iceberg-python/issues/208")
+    # except StopIteration:
+    #     pass
+    data_files = []
+    for task in tasks:
+        parquet_writer_kwargs = _get_parquet_writer_kwargs(table.properties)
```

Comment:
Both this one, and the `row_group_size` can be moved out of the for to avoid resolving all the properties with each task 👍